File: Backend/TripPlannerBackend.API/Entity/Trip.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Reflection.Metadata;
using System.Text;
using System.Threading.Tasks;

namespace TripPlannerBackend.API.Entity
{
    public class Trip
    {
        public int Id { get; set; }
        public TripType? Type { get; set; }
        public string Name { get; set; }
        public string Location { get; set; }
        public string Image { get; set; }
        public string? UserId { get; set; } = null;
        public DateTime Start { get; set; }
        public DateTime End { get; set; }
        public ICollection<Activity> Activities { get; set; }
        public bool IsPublic { get; set; } = false;
        public ICollection<User>? Users { get; set; }
        [ForeignKey("Cost")]
        public int? CostId { get; set; }
        public Cost? Cost { get; set; }
        public int Count { get; set; }

    }
}

File: Frontend/src/app/model/Trip.ts
import { Activity } from "./Activity";
import { Cost } from "./Cost";
import { TripType } from "./TripType";
import { User } from './User';

export interface Trip {
    id: number;
    name: string;
    type?: TripType;
    location: string;
    image: string | undefined;
    userId: string;
    users?: User[];
    start: Date;
    end: Date;
    cost?: Cost;
    activities: Activity[];
    isPublic: boolean;
    count: number;
}